### ISSUE TITLE
iOS: Processing non `RedrawRequested` event after the main event loop: AboutToWait

### ISSUE THREAD

**Issue by marcpabst** [rust-windowing/winit#3714]
### Description

Stuck on winit `0.29.15` for now, while waiting for the new API to take form. Running the event loop on iOS causes lots of warnings ````WARNING winit.platform_impl.platform.app_state 2024-05-28 23:23:27,897 app_state.rs:617 processing non `RedrawRequested` event after the main event loop: AboutToWait````. I'm also seeing sporadic slow downs.

I'm running winit through `CPython`/`Pyo3`, butt looks like there is a similar problem on using bevy on iPhone: https://github.com/bevyengine/bevy/discussions/12934

### Device and iOS version

iOS 15.0.2 (19A404)
iPad Pro (11-inch) (2nd generation)

### Winit version

0.29.15

**Comment by marcpabst** [comment]
Wuick follow-up (as I know this is hard to debug without code) - would it potentially help to just move to winit `0.30.*`? Have there been enough changes to event handling that this issue might just disappear on its own?

**Comment by madsmtm** [comment]
> would it potentially help to just move to winit `0.30.*`

Nope, it's an old issue, the application state tracking is broken on newer iOS versions.

**Comment by ActuallyHappening** [comment]
I am still getting this issue, with many noisy logs being printed every frame.
In the end I simply forked this from the point where the `crates.io` release was build, see here: https://github.com/ActuallyHappening/winit/tree/remove-noisy-logs

**Comment by marcpabst** [comment]
Do you think this is actually impacting permance or is log spam?

**Comment by madsmtm** [comment]
No idea, the backend is much older than my involvement in Winit, and haven't been updated for a long while. It's on my list to fix this, but said list is long!

**Comment by ActuallyHappening** [comment]
I might be wrong, but I don't think that specific log event is still around in the `main` branch / `0.30` release

**Comment by marcpabst** [comment]
Right, that inspires hope. I basically just noticed that while Metal is not the most reliably backend to begin with (in terms of dropped frames), I saw massive slow-downs on iPadOS that I wasn't seeing on MacOS.

**Comment by madsmtm** [comment]
> I might be wrong, but I don't think that specific log event is still around in the `main` branch / `0.30` release

It is:
https://github.com/rust-windowing/winit/blob/82d9bbe5598c8605a752641a477261de01340aed/src/platform_impl/apple/uikit/app_state.rs#L549

**Comment by ActuallyHappening** [comment]
I do not pretend to understand why this issue keeps coming up for me using `bevy`, would a PR that adds a 'hacky' #[cfg(not(winit_ignore_noisy_logs_unstable))]` on the offending `warn!(...)` be accepted?

**Comment by marcpabst** [comment]
I would be quite keen to understand any implication this might have before silencing the warning :)

**Comment by ActuallyHappening** [comment]
> I would be quite keen to understand any implication this might have before silencing the warning :)

What implications sorry?

My problem is my Xcode console I am getting a bunch of logs every frame. I've been slowly improving the quality of using `bevy` for iOS (see https://github.com/bevyengine/bevy/pull/13984), but the worst issue so far is the *spam* of events from winit (using `bevy 0.13` which depended on `winit 0.29.3` which used `log` not `tracing`, there was in fact no indication of where these logs were coming from since the log -> tracing compatibility infrastructure only renders the `event!` target as 'log' rather than 'winit::platform::ios'!)

Adding this hacky --cfg requirement would silence these warnings, since I don't know how to fix the underlying issue. A bandaid fix, if you will

**Comment by marcpabst** [comment]
A warning implies to me that something is actually _wrong_. It's usually better to actually fix the problem rather than to chose to ignore it. As I mentined, I saw slow-downs on iOS which might or might not be conencted to this.

**Comment by ActuallyHappening** [comment]
I agree.
